frontend/packages/ceph-storage-plugin/src/components/bs-ns-common/bs-ns-common.tsx

```diff
@@ -0,0 +1,266 @@
+import { useTranslation } from 'react-i18next';
+import { Button, FormGroup, TextInput, InputGroup } from '@patternfly/react-core';
+import { SecretModel } from '@console/internal/models';
+import { ResourceDropdown, getAPIVersion } from '@console/shared';
+import './create-bs-ns.scss';
+import { SecretType } from '@console/internal/components/secrets/create-secret';
+
+export enum StoreType {
+  BS = 'BackingStore',
+  NS = 'NamespaceStore',
+}
+export const awsRegionItems = _.zipObject(AWS_REGIONS, AWS_REGIONS);
+export const endpointSupported = [BC_PROVIDERS.S3, BC_PROVIDERS.IBM];
+export const getProviders = (type: StoreType) => {
+  const values =
+    type === StoreType.BS
+      ? Object.values(BC_PROVIDERS)
+      : Object.values(BC_PROVIDERS).filter(
+          (provider) => provider !== BC_PROVIDERS.GCP && provider !== BC_PROVIDERS.PVC,
+        );
+  return _.zipObject(values, values);
+};
+
+export const getExternalProviders = (type: StoreType) => {
+  return type === StoreType.NS
+    ? [BC_PROVIDERS.AWS, BC_PROVIDERS.AZURE, BC_PROVIDERS.S3, BC_PROVIDERS.IBM]
+    : [BC_PROVIDERS.AWS, BC_PROVIDERS.AZURE, BC_PROVIDERS.S3, BC_PROVIDERS.GCP, BC_PROVIDERS.IBM];
+};
+
+export type Payload = {
+  apiVersion: string;
+  kind: string;
+  metadata: {
+    namespace: string;
+    name: string;
+  };
+  spec: {
+    type: string;
+    ssl: boolean;
+    [key: string]: any;
+  };
+};
+
+export const secretPayloadCreator = (
+  provider: string,
+  namespace: string,
+  secretName: string,
+  field1: string,
+  field2 = '',
+) => {
+  const payload = {
+    apiVersion: getAPIVersion(SecretModel),
+    kind: SecretModel.kind,
+    stringData: {},
+    metadata: {
+      name: secretName,
+      namespace,
+    },
+    type: SecretType.opaque,
+  };
+
+  switch (provider) {
+    case BC_PROVIDERS.AZURE:
+      payload.stringData = {
+        AccountName: field1,
+        AccountKey: field2,
+      };
+      break;
+    case BC_PROVIDERS.IBM:
+      payload.stringData = {
+        IBM_COS_ACCESS_KEY_ID: field1,
+        IBM_COS_SECRET_ACCESS_KEY: field2,
+      };
+      break;
+    default:
+      payload.stringData = {
+        AWS_ACCESS_KEY_ID: field1,
+        AWS_SECRET_ACCESS_KEY: field2,
+      };
+      break;
+  }
+  return payload;
+};
+
+export type StoreAction =
+  | { type: 'setSecretName'; value: string }
+  | { type: 'setSecretKey'; value: string }
+  | { type: 'setAccessKey'; value: string }
+  | { type: 'setRegion'; value: string }
+  | { type: 'setTarget'; value: string }
+  | { type: 'setEndpoint'; value: string };
+
+export type ProviderDataState = {
+  secretName: string;
+  secretKey: string;
+  accessKey: string;
+  region: string;
+  target: string;
+  endpoint: string;
+};
+
+type S3EndpointTypeProps = {
+  type: StoreType;
+  state: ProviderDataState;
+  dispatch: React.Dispatch<StoreAction>;
+  provider: BC_PROVIDERS;
+  namespace: string;
+};
+
+export const S3EndPointType: React.FC<S3EndpointTypeProps> = (props) => {
```

Comment:
Why are we not removing it from the BS page? I thought we would make providers conditional based on a prop and make this component generic. What's the point of making it common and only using it in one place? The same applies to all the components here.